plugins/toolkit/jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/codescan/CodeWhispererCodeScanManager.kt

```diff
@@ -454,27 +459,55 @@ class CodeWhispererCodeScanManager(val project: Project) {
             val editorFactory = EditorFactory.getInstance()
             editorFactory.eventMulticaster.addDocumentListener(documentListener, project)
             editorFactory.addEditorFactoryListener(fileListener, project)
+
+            // Add document listeners to all files under project root for files of supported languages.
+            addListeners()
         }
     }
 
-    private fun addListeners() {
-        fileNodeLookup.keys.forEach { file ->
+    fun addListeners() {
+        val fileEditorManager = FileEditorManager.getInstance(project)
+        val files = fileEditorManager.openFiles
+        files.forEach { file ->
             runInEdt {
-                val document = FileDocumentManager.getInstance().getDocument(file)
-                if (document == null) {
-                    LOG.error { message("codewhisperer.codescan.file_not_found", file.path) }
-                    return@runInEdt
+                try {
+                    if (!file.isDirectory &&
+                        file.programmingLanguage().toTelemetryType() != CodewhispererLanguage.Unknown &&
+                        file.programmingLanguage().toTelemetryType() != CodewhispererLanguage.Plaintext
+                    ) {
+                        fileListenerLookup.add(file)
+                        file.getDocument().addDocumentListener(documentListener, project)
```

Comment:
what happens if an editor window is closed and opened again? wouldn't it be a different document instance?

Reply:
changed it to FileDocumentManager.getInstance().getDocument(virtualFile)